### PR TITLE
Bugfix/441 - Update data on network switch

### PR DESCRIPTION
**Pull request by ishrak-imam**
### Description
Appended the network name with the query key.

### CODE REVIEW

File: src/context/ChainApiContext.tsx [hunk 1]
```diff
@@ -30,7 +29,7 @@ export function ChainApiContextProvider({children}: {children: React.ReactNode})
       return;
     }
 
-    queryClient.clear();
+    // queryClient.clear();
```
Comment (ishrak-imam) [inline src/context/ChainApiContext.tsx]: We may not have to clear the cache on network switch now. Also it results to a better UX.
@franjs @ahoseinian please comment what you guys think.

File: src/api/hooks/useApiQuery.ts [hunk 2]
```diff
@@ -12,9 +12,10 @@ function useApiQuery<TData>(
   options: UseQueryOptions<TData> = {enabled: true},
 ) {
   const {api} = useApi();
+  const network = api?.runtimeChain != null ? api.runtimeChain.toString() : '';
 
   return useQuery(
-    queryKey,
+    `${queryKey}-${network}`,
```
Comment (ahoseinian) [inline src/api/hooks/useApiQuery.ts]: just one thing, the `queryKey` can be a string or array of string and objects. how about doing something like
```
[network, ...(typeof queryKey === 'string' ? [queryKey]: queryKey)]
```

File: src/context/ChainApiContext.tsx (hunk 1 above)
Comment (ahoseinian) [reply src/context/ChainApiContext.tsx]: yes I guess it would be unnecessary.

File: src/context/ChainApiContext.tsx [hunk 3]
```diff
@@ -30,7 +29,6 @@ export function ChainApiContextProvider({children}: {children: React.ReactNode})
       return;
     }
 
-    queryClient.clear();
```
Comment (silva-fj) [inline src/context/ChainApiContext.tsx:32]: `queryClient` is not used for anything else, can you remove the usage of `useQueryClient` as well?

File: src/api/hooks/useApiQuery.ts [hunk 4]
```diff
@@ -12,16 +12,22 @@ function useApiQuery<TData>(
   options: UseQueryOptions<TData> = {enabled: true},
 ) {
   const {api} = useApi();
+  const network = api?.runtimeChain != null ? api.runtimeChain.toString() : undefined;
```
Comment (silva-fj) [inline src/api/hooks/useApiQuery.ts:15]: You can also do: 
  `const {currentNetwork} = useContext(NetworkContext);
`

**Review by silva-fj — APPROVED** [review]

File: src/api/hooks/useApiQuery.ts (hunk 4 above)
Comment (ishrak-imam) [reply src/api/hooks/useApiQuery.ts:15]: `const {currentNetwork} = useContext(NetworkContext)`
I gave it a shot, but for some reason it does not behave as expected(refetch queries) for the screens in the drawer stack.

@franjs 
For now I am keeping it as it is. If you have time please ping me to discuss.

Comment (silva-fj) [reply src/api/hooks/useApiQuery.ts:15]: I'm not sure I follow, when we change the network we are actually changing the state in `NetworkContext` so it must be different, the issue is something else

**Pull request merged**